crates/algorithms/src/haversine/mod.rs

```diff
@@ -0,0 +1,131 @@
+use core::fmt::Debug;
+use nalgebra::Point2;
+use num_traits::Float;
+
+///calculate distance between point with use haversine formula
```

Comment:
This is an internal function, so it no longer needs documentation, and you did really good documentation in the macro implementation, so this can be removed^^